vendor/sigs.k8s.io/controller-reconciler/pkg/reconciler/manager/k8s/manager.go

```diff
@@ -116,6 +117,23 @@ func isReferringSameObject(a, b metav1.OwnerReference) bool {
 	return aGV == bGV && a.Kind == b.Kind && a.Name == b.Name
 }
 
+func isKindFormatUpdate(a, b metav1.OwnerReference) bool {
+	if a.APIVersion != b.APIVersion ||
+		a.Name != b.Name ||
+		a.UID != b.UID ||
+		getKindFromOldFormat(b.Kind) != a.Kind {
```

Comment:
// No need to update if these are two different kind 
if  ... !isKindEquivalent(a.kind, b.kind) {
 return false
}

if (a.kind != b.kind) {
return true;
}

return false

Reply:
Acknowledged